### PR TITLE
Enable MLIR by default in migraphx builds

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: rbuild.ini [hunk 1]
```diff
@@ -19,6 +19,7 @@ define =
 [develop]
 cxx = ${rocm_path}/llvm/bin/clang++
 cc = ${rocm_path}/llvm/bin/clang
+ignore = ROCmSoftwarePlatform/rocMLIR
```
Comment (umangyadav) [inline rbuild.ini:22]: Why does this one ignore it ? I use `rbuild develop` to install dependencies.

Comment (pfultz2) [reply rbuild.ini]: Yea this is the wrong place. It should be under the `gh` session.

**Comment by umangyadav** [comment]
what's the purpose behind having it enabled by default ? 
I see that we already have Jenkins job that builds MIGraphX with MLIR.

File: Dockerfile [hunk 2]
```diff
@@ -77,6 +77,9 @@ ADD dev-requirements.txt /dev-requirements.txt
 ADD requirements.txt /requirements.txt
 ADD rbuild.ini /rbuild.ini
 
+# Temporarily install a new cmake until switching to ubuntu 22.04
+RUN pip3 install cmake==3.22.1
```
Comment (causten) [inline Dockerfile:81]: ORT needs 3.26.4 would it make sense to bump this to 3.26.4 too?

Comment (pfultz2) [reply Dockerfile:81]: no, we should make sure our stuff builds with the cmake in ubuntu 22.04.

**Review by causten — APPROVED** [review]
builds and runs as expected.

**Comment by pfultz2** [comment]
The tidy error requires a fix in rocm-cmake: https://github.com/RadeonOpenCompute/rocm-cmake/pull/140

**Review by umangyadav — APPROVED** [review]

**Pull request merged**